doc/03_research/03_planning/Implementation.md

```diff
@@ -0,0 +1,135 @@
+# Planning Implementation
+
+**Summary:** 
+The document gives a first impression of how the planning could/should work
+and how the topics are edited
+---
+
+## Author
+
+Simon Erlbacher, Niklas Vogel
+
+## Date
+
+29.11.2022
+
+## Prerequisite
+
+
+---
+<!-- TOC -->
+* [Planning Implementation Details](#coding-style-guidelines)
+  * [Author](#author)
+  * [Date](#date)
+  * [Prerequisite](#prerequisite)
+  * [Visualization](#visualization)
+  * [Preplanning](#preplanning)
+  * [Local Path Planning](#local-path-planning)
+  * [Decision Making](#decision-making)
+  * [Next steps](#next-steps)
+* [Sources](#sources)
+<!-- TOC -->
+
+---
+## **Visualization** 
+
+![Implementation](../../00_assets/Planning_Implementierung.png)
+
+---
+## **Preplanning**
+  
+The Preplanning receives the next [target point](./basics.md#wie-sehen-die-daten-vom-leaderboard-fr-das-global-planning-aus) from the Carla Leaderboard. Also it reads the map data which is in [OpenDrive](https://www.asam.net/standards/detail/opendrive/) format. The [commonroad route planner (CRP)](https://gitlab.lrz.de/tum-cps/commonroad-route-planner/) from TUM can be used to calculate a Lanelet model (Note that almost all old PAF-Projects have adapted versions of the CRP). Internally the CRP converts the Map into a Scenario Object. When given a start and goal position and the current orientation of the vehicle a so called planning problem is created by the planer. The Orientation can be calculated on the Odometry and the GNUU data from the sensoring area. The Output (Solution of the planning problem) will be a route defined by a sequence of lanelets and a sequence of points (~ 10cm apart).
+
+Lanelet Model Example :
+![Lanelet Model Example](../../00_assets/Lanelets.png)[[Source]](https://github.com/ll7/psaf2/tree/main/Planning/global_planner)
+
+
+Input: 
+* Odometry data (sensoring)
+* GNUU data (sensoring)
+* Map
+* Navigation Waypoints
+
+Output:
+* Route (Sequences of Lanelets and Points) (local path planning, decision making)
+
+---
+
+## **Local Path Planning**
+
+Local Planner updates the current route, if Decision Making detects an obstacle. The planner then has to choose an alternative next Lanelet to avoid the obstacle. Therefore the obstacle position on the lanelet plan will be needed. 
+
+### <u>_Velocity profile_</u>
+
+The Local Path Planer receives the lanelets, points and the path to drive. The local planner creates a velocity profile on the calculated trajectory based on curvature, crossings and traffic lights. This will be calculated directly after the preplanning created a trajectory. The velocity value is published to the acting side, where the current speed and the braking distance is held and calculated. If the velocity is too high or too low, the Acting will realize and sends changes to adapt the speed of the vehicle.
```

Comment:
Eine Beschreibung der Funktionsweise des Acting ist hier nicht sinnvoll. Alles nach "The velocity value is published to the acting side" würde ich streichen und evtl. durch einen Verweis auf die Docs zu Acting ersetzten.